hax/helper/generate_sysconf.py

```diff
@@ -0,0 +1,159 @@
+        filepath = self.sysconfig_dir + filename
+        with open(filepath, 'w') as f:
+            f.write(contents)
+
+    def generate_confd(self, svc_id: str, hax_ep: str):
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        filename = f'm0d-{fid}'
+        contents = (f'MOTR_M0D_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n'
+                    'MOTR_CONF_XC=/etc/motr/confd.xc\n')
+        self._write_file(filename, contents)
+
+    def generate_ios(self, svc_id: str, hax_ep: str):
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        meta_data = self.get_ios_meta_data(svc_id)
+        filename = f'm0d-{fid}'
+        contents = (f'MOTR_M0D_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n')
+        if meta_data:
+            contents += f'MOTR_BE_SEG_PATH={meta_data}\n'
+        self._write_file(filename, contents)
+
+    def generate_s3(self, svc_id: str, hax_ep: str, s3_port: int):
+        profile_key = self.kv.kv_get('m0conf/profiles/', keys=True)[0]
+        assert profile_key
+        profile_fid = profile_key.split('/')[-1]
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        filename = f's3server-{fid}'
+        contents = (f'MOTR_PROFILE_FID={profile_fid}\n'
+                    f'MOTR_S3SERVER_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n'
+                    f'MOTR_S3SERVER_PORT={s3_port}\n')
+        self._write_file(filename, contents)
+
+    def generate_sysconfig(self, IDs: Dict):
+        s3_port = 28071
+        hax_ep = self.get_service_ep(IDs['HAX_ID'][0])
+
+        for x in IDs['CONFD_IDs']:
+            self.generate_confd(x, hax_ep)
+
+        for x in IDs['IOS_IDs']:
+            self.generate_ios(x, hax_ep)
+
+        for x in IDs['S3_IDs']:
+            self.generate_s3(x, hax_ep, s3_port)
+            s3_port += 1
+
+    def get_all_svc_ids(self) -> Dict[str, List[str]]:
+
+        IDs: Dict[str, List[str]] = {}
+        IDs['HAX_ID'] = self.get_service_ids('ha')
+        IDs['CONFD_IDs'] = self.get_service_ids('confd')
+        IDs['IOS_IDs'] = self.get_service_ids('ios')
+        IDs['S3_IDs'] = self.get_service_ids('m0_client_s3')
+        return IDs
+
+
+def main():
+    _setup_logging()
+    try:
+        generator = Generator()
+        IDs = generator.get_all_svc_ids()
+        if not IDs['HAX_ID']:
+            raise RuntimeError(
+                'Cannot get information about Hax from Consul for this host '
+                f'{get_node_name()}. Please verify that the host name '
+                'matches the one stored in the Consul KV.')
+        generator.generate_sysconfig(IDs)
+    except Exception as e:
+        logging.error('Exiting due to a failure: %s', e)
```

Comment:
I think it makes sense to exit with non-zero code if an exception is catched here.

Reply:
> I think it makes sense to exit with non-zero code if an exception is catched here.

Addressed.